### PR TITLE
Workaround MSVC bug that std::isnan cannot handle integral type

### PR DESCRIPTION
**Pull request by rongjiecomputer**
#15213

### CODE REVIEW

**Review by jlebar — COMMENTED** [review]
In addition this is good because we don't want to call std::fmin on an integral type!  That loses precision for 64-bit integers due to the cast to double.

Perhaps it would be worth writing a quick testcase for that, since (I think) you're fixing an actual bug here?

**Comment by rongjiecomputer** [comment]
@jlebar Is this the right place to add the test?
https://github.com/tensorflow/tensorflow/blob/70298632d0e4e2b5cd3b9270dd44b2d1bb24a874/tensorflow/compiler/xla/service/hlo_evaluator_test.cc#L134

File: tensorflow/compiler/xla/service/hlo_evaluator_test.cc [hunk 1]
```diff
@@ -152,6 +152,33 @@ TEST_P(HloEvaluatorBf16Test, DoesClamp) {
   EXPECT_TRUE(LiteralTestUtil::Equal(expected, result));
 }
 
+// Verifies that clamping of int64 does not cause loss of precision
+TEST_P(HloEvaluatorBf16Test, DoesClamp) {
+  auto ones = [](int bits) -> int64 { return (1LL << bits) - 1; };
```
Comment (jlebar) [inline tensorflow/compiler/xla/service/hlo_evaluator_test.cc]: Nit, suggest s/1LL/int64{1}/ so we're not assuming `long long == int64`.

Comment (jlebar) [inline tensorflow/compiler/xla/service/hlo_evaluator_test.cc]: Nit, we usually leave out the `-> T` when it's obvious, as it is here.

**Review by jlebar — DISMISSED** [review]
LGTM!  Just two nits.

(I'm hitting approve now, but I think I'll have to re-approve after your changes for it to be pulled in.)

**Review by jlebar — APPROVED** [review]
Approved!  Thank you for the patch!

I'll put this in the queue to get checked in.  It may take a few days, esp since it's Friday afternoon.

**Comment by rongjiecomputer** [comment]
Ping. Test failures should be unrelated to my change, do I need to rebase?

**Comment by jlebar** [comment]
@hgadig I believe this is ready to pull?

**Comment by hgadig** [comment]
> @hgadig I believe this is ready to pull?

@jlebar  Looks like the CL has merge conflicts. Request you to take a look.

**Comment by jlebar** [comment]
> Request you to take a look.

Sure, since this appears to be our mistake that this sat for two weeks with no response, I'd like to try merging this PR without asking the contributor to rebase.

What's the best way to do this?

**Comment by jlebar** [comment]
@hgadig friendly ping

**Comment by jlebar** [comment]
@hgadig FYI I will be OOO tomorrow through Tuesday, so if we can't fix this today, please send the internal review to timshen or sanjoy.

**Comment by hgadig** [comment]
> @hgadig FYI I will be OOO tomorrow through Tuesday, so if we can't fix this today, please send the internal review to timshen or sanjoy.

Sure. I'm working with Yifei to resolve internal merge conflicts. Thanks !

**Comment by jlebar** [comment]
woohoo, thank you for your patience here.  Very happy this one is done.

**Pull request merged**